src/providers/ogr/qgsogrprovider.h

```diff
@@ -212,6 +298,49 @@ class QgsOgrProvider : public QgsVectorDataProvider
     //! String used to define a subset of the layer
     QString mSubsetString;
 
+    /** MSubLayerList entry that was actually used to open the layer
```

Comment:
That is always a problem with documentation
* either to much, or not enough (mostly the later)

Due to the complexity of the matter it should be done properly as a whole (overview) somewhere
* **QgsOgrProviderUtils::OGRGetSubLayersWrapper**
   * is, I think, the best place since the function deal with all aspects

So a Overview/Summery should exist as a reference
* to understand better why something is being done in certain way

**Note**: most of this was written around September last year
* intended for QGIS 2.18 with gdal 1

I have removed most of those, but there are still leftovers